Subscription Management: Redirect /following/manage to /read/subscriptions

TimBroddin opened the pull request:
Related to #78796

## Proposed Changes

Redirects visitors of `/following/manage` to `/read/subscriptions`

## Testing Instructions

- Apply this PR
- Visit `http://calypso.localhost:3000/following/manage`
- You should be redirected to `/read/subscriptions`

This should only happen in the dev environment. To try it in other environments, append `?flags=subscription-management-redirect-following`

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

TimBroddin committed:
Redirect /following/manage to /read/subscriptions

yansern commented:
Thanks for the quick work! Could this be a conditional that's behind a feature flag and less destructive?

TimBroddin committed:
Add 'subscription-management-redirect-following' feature flag

A new feature flag 'subscription-management-redirect-following' has been added to all configuration files. This flag allows us to toggle redirection behavior for the subscription management feature. It's initially set to 'false' in all environments except development.json, where it's set to 'true' for testing purposes. Concurrent changes have been made in 'client/reader/following/controller.js' to incorporate the behavior driven by this flag.

mashikag commented on client/reader/following/controller.js:
```diff
@@ -7,31 +9,35 @@ const analyticsPageTitle = 'Reader';
+				/>
+			);
+		}
 		next();
```
It looks to me like the `next();` can also be included in the else, and/or you can simply return on line 13, this would save the need for the explicit `else` statement too.

mashikag committed:
Return at page redirect

mashikag reviewed (approved):
Tested, and all looks good to me. 👌
